database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -358,4 +362,51 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
       labelAndKeyLabelValues, labelOrKeyLabelValues, excludeLabelAndValues, excludeLabelOrValues, submissionTimestampOption, startTimestampOption, endTimestampOption, metadataArchiveStatus, includeSubworkflows)
     runTransaction(action)
   }
+
+  override def deleteNonLabelMetadataForWorkflow(rootWorkflowId: String)(implicit ec: ExecutionContext): Future[Int] = {
+
+    val delete: DBIO[Int] = checkDeletionPreconditions(rootWorkflowId) flatMap { checkResult: Checked[Unit] =>
```

Comment:
Again, the `database` is supposed to only contain scala-to-sql translation-- with little to zero business-logic and error-handling.

There are two directions we could go here:
- Move this logic check up into the services layer (there are examples of passing in functions, but those are for more complex cases), or
- Preferred and simpler, move this check to run down _inside the database_

Reply:
>Preferred and simpler, move this check to run down inside the database

Will give this a go, should be relatively sane with the test suite in place. Thanks!